From f6befbcfe50d37932b33316f078cc08cc76e1a9e Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 12 May 2020 20:11:14 +0530
Subject: [PATCH 0499/1921] octeontx2-af: Fix nix_inline_ipsec_cfg mailbox

nix_inline_ipsec_cfg may be called by AF consumer
before any LF is attached to it hence call
rvu_get_blkaddr accordingly.

Fixes: 3bb5a5
("octeontx2-af: Return assigned NIX/CPT block address")

Change-Id: Ibf1e005f311ec4696c6aafe916cf079787aeecd1
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/28282
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 2866ef702252..4269656900d5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -4376,7 +4376,7 @@ int rvu_mbox_handler_nix_inline_ipsec_cfg(struct rvu *rvu,
 	if (!is_block_implemented(rvu->hw, BLKADDR_CPT0))
 		return 0;
 
-	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_NIX, req->hdr.pcifunc);
+	blkaddr = rvu_get_blkaddr(rvu, BLKTYPE_NIX, 0);
 	if (blkaddr < 0)
 		return NIX_AF_ERR_AF_LF_INVALID;
 
-- 
2.31.1

